### PR TITLE
fix(ktreelist): add group prop [KHCP-13028]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-13028

Adds `group` prop in KTreeList that allows disabling dragging items from one list into another

### CODE REVIEW

File: sandbox/pages/SandboxTreeList.vue [hunk 1]
```diff
@@ -12,28 +12,43 @@
       <SandboxSectionComponent title="item">
         <KTreeList :items="items1" />
       </SandboxSectionComponent>
+      <SandboxSectionComponent title="group">
```
Comment (adamdehaven) [inline sandbox/pages/SandboxTreeList.vue:15]: Can you update the example to have 2 sets of treelists like on the docs example?

File: docs/components/tree-list.md [hunk 2]
```diff
@@ -147,6 +147,30 @@ const items = ref<TreeListItem[]>([
...
+
+In the example below, try toggling grouping on and off and dragging items from one list into another.
+
+<KComponent
+  v-slot="{ data }"
+  :data="{ grouping: true }"
+>
+  <KInputSwitch
+    v-model="data.grouping"
+    label="Grouping"
```
Comment (adamdehaven) [inline docs/components/tree-list.md:162]: "Grouping" is a bit confusing, can you change to "Same group" and "Different groups"?
```suggestion
    :label="data.grouping ? 'Same group' : 'Different groups'"
```

File: sandbox/pages/SandboxTreeList.vue (hunk 1 above)
Comment (portikM) [reply sandbox/pages/SandboxTreeList.vue:15]: Done

File: docs/components/tree-list.md (hunk 2 above)
Comment (portikM) [reply docs/components/tree-list.md]: Done

File: docs/components/tree-list.md [hunk 3]
```diff
@@ -147,6 +147,30 @@ const items = ref<TreeListItem[]>([
...
+
+In the example below, try toggling grouping on and off and dragging items from one list into another.
+
+<KComponent
+  v-slot="{ data }"
+  :data="{ grouping: true }"
+>
+  <KInputSwitch
+    v-model="data.grouping"
+    :label="data.grouping ? 'Same group' : 'Different groups'"
```
Comment (adamdehaven) [inline docs/components/tree-list.md:162]: My fault, I think this is backwards

File: docs/components/tree-list.md [hunk 4]
```diff
@@ -147,6 +147,30 @@ const items = ref<TreeListItem[]>([
 </script>
 ```
 
+### group
+
+To drag elements from one list into another, both lists must have the same `group` value. Defaults to `k-tree-list` (meaning that elements from one list on a page can be dragged into another, unless a different `name` value is provided for one of them).
+
+In the example below, try toggling grouping on and off and dragging items from one list into another.
```
[the remark this answers is not in the document]
Comment (portikM) [reply docs/components/tree-list.md:154]: Done

File: docs/components/tree-list.md (hunk 3 above)
Comment (portikM) [reply docs/components/tree-list.md]: That's what I thought -- fixed

File: docs/components/tree-list.md [hunk 5]
```diff
@@ -147,6 +147,30 @@ const items = ref<TreeListItem[]>([
 </script>
 ```
 
+### group
+
+To drag elements from one list into another, both lists must have the same `group` value. Defaults to `k-tree-list` (meaning that elements from one list on a page can be dragged into another, unless a different `name` value is provided for one of them).
```
[the remark this answers is not in the document]
Comment (portikM) [reply docs/components/tree-list.md:152]: Yes, thank you

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**